pkg/alexa/l10n/l10n.go

```diff
@@ -0,0 +1,224 @@
+	GetAny(key string, args ...interface{}) string
+	GetAll(key string, args ...interface{}) []string
+}
+
+// Config contains the options for Locale registration
+type Config struct {
+	DefaultLocale bool
+	FallbackFor   string
+}
+
+// TODO: move to `ssml` package
+func Speak(text string) string {
+	return "<speak>" + text + "</speak>"
+}
+func UseVoice(voice string, text string) string {
+	return `<voice name="` + voice + `">` + text + `</voice>`
+}
+
+func UseVoiceLang(voice string, language string, text string) string {
+	return `<voice name="` + voice + `"><lang xml:lang="` + language + `">` + text + `</lang></voice>`
+}
+
+// DefaultRegistry is the standard registry used
+var DefaultRegistry = &Registry{
+	locales: map[string]LocaleInstance{},
+}
+
+// Registry is the Locale registry
+type Registry struct {
+	defaultLocale string
+	locales       map[string]LocaleInstance
+}
+
+func NewRegistry() LocaleRegistry {
+	return &Registry{locales: map[string]LocaleInstance{}}
+}
+
+// RegisterFunc defines the functions to be passed to Register
+type RegisterFunc func(cfg *Config)
+
+// AsDefault sets the given Locale the default
+func AsDefault() RegisterFunc {
+	return func(cfg *Config) {
+		cfg.DefaultLocale = true
+	}
+}
+
+// Register registers a new Locale in the DefaultRegistry
+func Register(locale LocaleInstance, opts ...RegisterFunc) error {
+	return DefaultRegistry.Register(locale, opts...)
+}
+
+// Resolve returns the matching Locale from the DefaultRegistry
+func Resolve(name string) (LocaleInstance, error) {
+	return DefaultRegistry.Resolve(name)
+}
+
+// Register registers a new locale and fails if it already exists
+func (r *Registry) Register(l LocaleInstance, opts ...RegisterFunc) error {
+	_, ok := r.locales[l.GetName()]
+	if ok {
+		return fmt.Errorf("locale %s already registered", l.GetName())
+	}
+
+	// run all RegisterFuncs
+	var cfg Config
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+
+	// set locale as default
+	if cfg.DefaultLocale {
+		r.defaultLocale = l.GetName()
+	}
+
+	r.locales[l.GetName()] = l
+
+	return nil
+}
+
+func (r *Registry) GetDefault() LocaleInstance {
+	return r.locales[r.defaultLocale]
+}
+func (r *Registry) GetLocales() map[string]LocaleInstance {
+	return r.locales
+}
+
+// Resolve returns the Locale matching the given name or an error
+func (r *Registry) Resolve(locale string) (LocaleInstance, error) {
+	l, ok := r.locales[locale]
+	if !ok {
+		return nil, fmt.Errorf("locale %s not found", locale)
+	}
+	return l, nil
+}
+
+///////////////////////////////////////////
+
+// Locale is a representation of keys in a specific language (and can have a fallback Locale)
+type Locale struct {
+	Name         string          // de-DE, en-US, ...
+	Countries    []alexa.Country // countries associated with this locale
+	TextSnippets Snippets
+}
+
+func NewLocale(locale string) *Locale {
+	return &Locale{
+		Name:         locale,
+		TextSnippets: Snippets{},
+	}
+}
+
+func (l *Locale) GetName() string {
```

Comment:
Name isnt private, you dont need this.

Reply:
GetName and Set are required by the interface, but I also want to allow building a `Locale` object manually
```
var foo := &l10n.Locale{
  Name: "en-US",
}
```